plugins/parquet/parquetembed.cpp

```diff
@@ -0,0 +1,1886 @@
+ *
+ * @param value Data to be written to the parquet file.
+ * @param field Object with information about the current field.
+ */
+void ParquetRecordBinder::processInt(__int64 value, const RtlFieldInfo *field)
+{
+    bindIntParam(value, field, r_parquet);
+}
+
+/**
+ * @brief Calls the bind function for the data type of the value.
+ *
+ * @param value Data to be written to the parquet file.
+ * @param field Object with information about the current field.
+ */
+void ParquetRecordBinder::processUInt(unsigned __int64 value, const RtlFieldInfo *field)
+{
+    bindUIntParam(value, field, r_parquet);
+}
+
+/**
+ * @brief Calls the bind function for the data type of the value.
+ *
+ * @param value Data to be written to the parquet file.
+ * @param field Object with information about the current field.
+ */
+void ParquetRecordBinder::processReal(double value, const RtlFieldInfo *field)
+{
+    bindRealParam(value, field, r_parquet);
+}
+
+/**
+ * @brief Calls the bind function for the data type of the value.
+ *
+ * @param value Data to be written to the parquet file.
+ * @param digits Number of digits in decimal.
+ * @param precision Number of digits of precision.
+ * @param field Object with information about the current field.
+ */
+void ParquetRecordBinder::processDecimal(const void *value, unsigned digits, unsigned precision, const RtlFieldInfo *field)
+{
+    Decimal val;
+    size32_t bytes;
+    rtlDataAttr decText;
+    val.setDecimal(digits, precision, value);
+    val.getStringX(bytes, decText.refstr());
+
+    bindDecimalParam(decText.getstr(), field, r_parquet);
+}
+
+/**
+ * @brief Calls the bind function for the data type of the value.
+ *
+ * @param chars Number of chars in the value.
+ * @param value Data to be written to the parquet file.
+ * @param field Object with information about the current field.
+ */
+void ParquetRecordBinder::processUnicode(unsigned chars, const UChar *value, const RtlFieldInfo *field)
+{
+    bindUnicodeParam(chars, value, field, r_parquet);
+}
+
+/**
+ * @brief Calls the bind function for the data type of the value.
+ *
+ * @param len Length of QString
+ * @param value Data to be written to the parquet file.
+ * @param field Object with information about the current field.
+ */
+void ParquetRecordBinder::processQString(unsigned len, const char *value, const RtlFieldInfo *field)
+{
+    size32_t charCount;
+    rtlDataAttr text;
+    rtlQStrToStrX(charCount, text.refstr(), len, value);
+
+    bindStringParam(charCount, text.getstr(), field, r_parquet);
+}
+
+/**
+ * @brief Calls the bind function for the data type of the value.
+ *
+ * @param chars Number of chars in the value.
+ * @param value Data to be written to the parquet file.
+ * @param field Object with information about the current field.
+ */
+void ParquetRecordBinder::processUtf8(unsigned chars, const char *value, const RtlFieldInfo *field)
+{
+    bindStringParam(chars, value, field, r_parquet);
```

Comment:
This needs to call a bindUtfParameter - which performs no translation

Reply:
Created bindUtf8Param().